### PR TITLE
Add Settings window content and functionality

### PR DESCRIPTION
**Pull request by nicolasespinoza**
### Add Settings window content and functionality
- Each respective page is populated with relevant settings
- Controls can be customized with "Press Any Key" capabilities
- Added default preferences and controls

Please note this does not include any backend functionality, it only updates the PreferenceManager to be accessed later.

**General page**
settings-general

**Controls page**
settings-controls

### CODE REVIEW

File: modules/SynthesisCore/UI/Windows/SettingsWindow.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                        PreferenceManager.SetPreference("SynthesisCore", key, PendingChanges[key]);
+                    }
+                }
+                PreferenceManager.Save();
+                
+                UIManager.ClosePanel(Panel.Name);
+            });
+
+            Button closeButton = (Button) Window.Get("close-button");
+            closeButton?.Subscribe(x =>
```
[the remark this answers is not in the document]
Comment (nicolasespinoza) [reply modules/SynthesisCore/UI/Windows/SettingsWindow.cs:92]: Yes, I agree. That should be included in the latest commit

**Review by copernicium — APPROVED** [review]
After some talk, we decided the settings panel doesn't need an "Apply" button.

**Review by KyroVibe — APPROVED** [review]
Looks good. UI flows pretty good and the key detection is nice

**Comment by copernicium** [comment]
@matthew17754 Is this ready to merge? Or do you still want to take a look first?

**Review by matthew17754 — APPROVED** [review]
All forms have dragability enabled. Preferences save properly. UI functions correctly. It'd be nice to specify that `Quality` represents `Graphics Quality`. Other than that, everything looks good.

**Pull request merged**